### ISSUE TITLE
Getting Things Done (GTD) Methodology Incompatibility

### ISSUE THREAD

**Issue by RWSlot** [johannesjo/super-productivity#2920]

**Comment by johannesjo** [comment]
Can you elaborate? :)

**Comment by RWSlot** [comment]
I think people are working on this issue. As I explored the GitHub more thoroughly, it looks like it's been a topic of conversation for a while.

**Comment by mathisgauthey** [comment]
The only flaw I see concerning GTD here is that we can't reorganize tags or only show tags categories that currently have tasks inside.

The best design for a GTD would be to have : 
- Some fixed tags on the sidebar : `next`, `waiting`, `delegated` and maybe `agenda`
- Other tags would just be shown on the sidebar **only** if an active task is tagged, and they would be context tags such as `errands`, `phone`, `computer`, `transports`, etc.
- The ability to organize tags : https://github.com/johannesjo/super-productivity/issues/2895

Basically, we miss the ability to pin tags and have some appears in the sidebar only when active tasks are tagged. And we miss the reorganization. That's why I personally use Obsidian for my GTD setup.

**Comment by johannesjo** [comment]
@mathisgauthey Thanks for the input. Adding a "pin tag" feature shouldn't be too complicated. I am all for it, since it seems useful even beyond GTD.

I am less sure about the dynamic tags though. Could you explain to me why this would be necessary?

**Comment by RWSlot** [comment]
Wow, that's pretty incredible. I'm just starting to use Obsidian and can see the potential. I was thinking of using Obsidian for my GTD as well, but am still learning how it works. I would love to witness part of your setup. Do you have your methodology or a demo posted somewhere?

**Comment by mathisgauthey** [comment]
> Wow, that's pretty incredible. I'm just starting to use Obsidian and can see the potential. I was thinking of using Obsidian for my GTD as well, but am still learning how it works. I would love to witness part of your setup. Do you have your methodology or a demo posted somewhere?

Hey there, you can find my "old" setup here from mid-2023 : https://github.com/mathisgauthey/obsidian-workflow-template

I have now updated (and simplified) it quite a bit, but didn't yet share the changes as I'm looking for a job.

**Comment by Jagdfalke** [comment]
> > @mathisgauthey Thanks for the input.! Adding a "pin tag" feature shouldn't be too complicated. I am all for it, since it seems useful even beyond GTD.
> > I am less sure about the dynamic tags though. Could you explain to me why this would be necessary?
> 
> Sure.
> 
> Let's take Todoist as an example as it was my past setup. I needed to add my basics GTD filters such as `next`, `waiting_for` and `delegated`. There's also `agenda` for tasks or things I need to discuss with someone in particular next time I see them.
> 
> And then I'll need to manually add tags for each contexts (related to `next` tags) :
> 
> * Errands
> * Computer
> * Phone
> * Transports
> * etc
> 
> And for each people (related to `delegated` or `agenda`) :
> 
> * Boss
> * Coworker 1
> * Coworker 2
> * Girlfriend
> * etc
> 
> > Then I'll query tasks to see my next actions grouped by context, or my agenda/delegated ones grouped by people.
> 
> The issue is that **each time I need to add or remove a context in my life, I need to create or remove tags in my todoist config :**
> 
> ![image](https://private-user-images.githubusercontent.com/46576952/293914505-5c732ef0-e1a2-4c06-b3c1-4c01f9cf4e63.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NTMzOTksIm5iZiI6MTcxMTY1MzA5OSwicGF0aCI6Ii80NjU3Njk1Mi8yOTM5MTQ1MDUtNWM3MzJlZjAtZTFhMi00YzA2LWIzYzEtNGMwMWY5Y2Y0ZTYzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDE5MTEzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM2ZTZmMDgwZmQwZjdmMmUwZjJhODQ2ZmM1ZTc4OTMyYmJjZmQ3Y2M5ZDQ5MDQ3ZWU1OWMzZTlhYjkxNDNjZmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Am6H7SMPrCJHBs8w4JYR8zlKlcc2IkZ3PAumJ-nx2So)
> 
> Which is the same issue with SupProd.
> 
> Now with Obsidian, I have a single task query that takes every tasks with a `#next` tag, and group them by the other associated tag (which is a context) :
> 
> ```
> > [!IMPORTANT]+ Next Actions 🏃
> >
> > ```tasks
> > not done
> > description does not include ]]
> > tags include #next
> > group by function task.tags.filter( (tag) => ! tag.includes("#next") )
> > ```
> ```
> 
> Here's the result of this query : ![image](https://private-user-images.githubusercontent.com/46576952/293913738-412dbc4d-82b2-422c-9223-80b49f9492f0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NTMzOTksIm5iZiI6MTcxMTY1MzA5OSwicGF0aCI6Ii80NjU3Njk1Mi8yOTM5MTM3MzgtNDEyZGJjNGQtODJiMi00MjJjLTkyMjMtODBiNDlmOTQ5MmYwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDE5MTEzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMwY2VlOWI1ZmIzYjRmZGQ0YWVlOGNhNjFhYjhlZGJlMTcyNWM2OWY0YzQyMTMxYmY5NDcyYTRjYmZmOWJmZjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.CVos4Zrk37NAmjLrdkna7NppDpGBNhdkqu_J_5bdgPE)
> 
> You could easily imagine the same setup for `delegated` or `agenda` tasks items using a similar query.
> 
> The main advantage is that when writing a task, **I don't need to figure out if this context or person tag already exists. If it does, I'll be suggested by the auto-completer. If it doesn't, I create it and I'll be able to see it in my queries as long as some tasks have this context.**
> 
> There is no manual manipulation of context and agenda tags as they may (and will) come and go with life. The only thing that truly needs to be static are the core tags of the GTD setup, which are :
> 
> * next
> * waiting_for
> * delegated
> * agenda
> 
> Hope it helps o/

Might I suggest an alternative solution? What I understand is that for GTD you need your core tags: next, waiting, delegated and agenda. These you could use in Super Productivity. Whenever you require a certain context (e.g. house, van, mail) you could use Projects in Super Productivity. This will enable you to keep your list of tags short, maintain the context as well as enable you to sort everything by context.

**Comment by mathisgauthey** [comment]
> > > @mathisgauthey Thanks for the input.! Adding a "pin tag" feature shouldn't be too complicated. I am all for it, since it seems useful even beyond GTD.
> > > I am less sure about the dynamic tags though. Could you explain to me why this would be necessary?
> > 
> > Sure.
> > 
> > Let's take Todoist as an example as it was my past setup. I needed to add my basics GTD filters such as `next`, `waiting_for` and `delegated`. There's also `agenda` for tasks or things I need to discuss with someone in particular next time I see them.
> > 
> > And then I'll need to manually add tags for each contexts (related to `next` tags) :
> > 
> > * Errands
> > * Computer
> > * Phone
> > * Transports
> > * etc
> > 
> > And for each people (related to `delegated` or `agenda`) :
> > 
> > * Boss
> > * Coworker 1
> > * Coworker 2
> > * Girlfriend
> > * etc
> > 
> > > Then I'll query tasks to see my next actions grouped by context, or my agenda/delegated ones grouped by people.
> > 
> > The issue is that **each time I need to add or remove a context in my life, I need to create or remove tags in my todoist config :**
> > 
> > ![image](https://private-user-images.githubusercontent.com/46576952/293914505-5c732ef0-e1a2-4c06-b3c1-4c01f9cf4e63.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NTMzOTksIm5iZiI6MTcxMTY1MzA5OSwicGF0aCI6Ii80NjU3Njk1Mi8yOTM5MTQ1MDUtNWM3MzJlZjAtZTFhMi00YzA2LWIzYzEtNGMwMWY5Y2Y0ZTYzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDE5MTEzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM2ZTZmMDgwZmQwZjdmMmUwZjJhODQ2ZmM1ZTc4OTMyYmJjZmQ3Y2M5ZDQ5MDQ3ZWU1OWMzZTlhYjkxNDNjZmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Am6H7SMPrCJHBs8w4JYR8zlKlcc2IkZ3PAumJ-nx2So)
> > 
> > Which is the same issue with SupProd.
> > 
> > Now with Obsidian, I have a single task query that takes every tasks with a `#next` tag, and group them by the other associated tag (which is a context) :
> > 
> > ```
> > > [!IMPORTANT]+ Next Actions 🏃
> > >
> > > ```tasks
> > > not done
> > > description does not include ]]
> > > tags include #next
> > > group by function task.tags.filter( (tag) => ! tag.includes("#next") )
> > > ```
> > ```
> > 
> > Here's the result of this query : ![image](https://private-user-images.githubusercontent.com/46576952/293913738-412dbc4d-82b2-422c-9223-80b49f9492f0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NTMzOTksIm5iZiI6MTcxMTY1MzA5OSwicGF0aCI6Ii80NjU3Njk1Mi8yOTM5MTM3MzgtNDEyZGJjNGQtODJiMi00MjJjLTkyMjMtODBiNDlmOTQ5MmYwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDE5MTEzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMwY2VlOWI1ZmIzYjRmZGQ0YWVlOGNhNjFhYjhlZGJlMTcyNWM2OWY0YzQyMTMxYmY5NDcyYTRjYmZmOWJmZjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.CVos4Zrk37NAmjLrdkna7NppDpGBNhdkqu_J_5bdgPE)
> > 
> > You could easily imagine the same setup for `delegated` or `agenda` tasks items using a similar query.
> > 
> > The main advantage is that when writing a task, **I don't need to figure out if this context or person tag already exists. If it does, I'll be suggested by the auto-completer. If it doesn't, I create it and I'll be able to see it in my queries as long as some tasks have this context.**
> > 
> > There is no manual manipulation of context and agenda tags as they may (and will) come and go with life. The only thing that truly needs to be static are the core tags of the GTD setup, which are :
> > 
> > * next
> > * waiting_for
> > * delegated
> > * agenda
> > 
> > Hope it helps o/
> 
> Might I suggest an alternative solution? What I understand is that for GTD you need your core tags: next, waiting, delegated and agenda. These you could use in Super Productivity. Whenever you require a certain context (e.g. house, van, mail) you could use Projects in Super Productivity. This will enable you to keep your list of tags short, maintain the context as well as enable you to sort everything by context. 

Hey there, thanks for the input. Unfortunately, I need to be able to have Next actions from multiple projects, while not moving them around. Basically tags serves as filters to work on relevant things based on what is around me (Next actions with context like a computer or during public transport, agenda with people like my manager, waiting for and delegated for things I'm waiting on something to happen before doing something else).

Hence it can't be a project, I need to be able to have todos live inside a project and query them according to their tags and dates. A task can then be at multiple places at a time (my agenda, my next action list, etc).

That's a workflow I'm doing easily using Ticktick right now but I would much prefer a more open-sourced approach 👌

**Comment by mathisgauthey** [comment]
Still relevant 👌

**Comment by franco-ruggeri** [comment]
I use GTD too. Although I have implemented it in SuperProductivity with some tradeoffs, from my perspective, the main missing feature is filters. More specifically, since in GTD there are 4 dimensions to choose what to do (context, time, energy, and priority), it should be possible to select tasks having a logical AND of conditions on tags and estimated time. For example:
- If I'm at the computer in the morning with full energy, I need to select tasks with tags `at computer`, `high energy`, and `high priority`. 
- If I have a meeting in 20 minutes, I want to select tasks with tags `at computer`, `low energy`, `high priority`, and `estimated time < 20m`.

For reference, I currently implement GTD as follows:
<details>
  <summary>GTD in SuperProductivity</summary>

- Project lists: task lists with `p/` prefix.
- Project support material (planned tasks): project task list (`p/` prefix).
- Time-specific actions:
	- Today: `Today` tag.
	- Future days: backlog + schedule.
- Next action lists: `a/*` tags (a = action).
- Waiting for list: `waiting-for` tag + `Wait for <name> - ` prefix.
- Someday list: `someday` tag + backlog.
- Tickler file: `someday` tag + schedule + backlog.
- Four-criteria model for choosing the action in the moment:
	- Context: `a/*`.
	- Time: estimated time. This feature also implements time boxing.
	- Energy: `e/low` and `e/high` (e = energy).
	- Priority: `p/high`, `p/medium`, `p/low` (p = priority). This scheme also implements the ABC Method.
</details>

**Comment by johannesjo** [comment]
If you care about this feature please upvote the original issue. This helps me understand how popular this request is.

**Comment by franco-ruggeri** [comment]
> I use GTD too. Although I've been using it in SuperProductivity with some tradeoffs, from my perspective, the main missing feature is having flexible filters (by tag and by estimated time). More specifically, since in GTD there are 4 dimensions to choose which task to do (context, time, energy, and priority), it should be possible to select tasks having a logical AND of conditions on tags and estimated time. For example:
> 
>     * If I'm at the computer in the morning with full energy, I need to select tasks with tags `at computer`, `high energy`, and `high priority`.
> 
>     * If I have a meeting in 20 minutes, I want to select tasks with tags `at computer`, `low energy`, `high priority`, and `estimated time < 20m`.
> 
> 
> For reference, I currently implement GTD as follows:
> My GTD in SuperProductivity

The new boards feature is very useful to achieve this. It can be used to select tasks with a combination of tags (e.g. `at computer`, `low energy`, `high priority`). The missing thing is to let users also sort by estimated time.

**Comment by JcMinarro** [comment]
> I think this should be solved now for the most part by the new boards feature?

The new board looks great. 
I think would be useful for GTD to have: 
- Order by. Allowing to sort the list within a panel by estimated time
- Group by. Allowing to group the list within a panel by a list of predefined tag. Let say a panel is configured to filter tasks with label "next", the config section could provide a new list of tags which will be used to group them. If the "next" labeled task contain any of the tag on the "group by" config, they are grouped there, if none of the rest tags is contained in "group by" config, an extra group for them is created

**Comment by johannesjo** [comment]
@JcMinarro can you open up a new issue for this?
